test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```diff
@@ -319,6 +345,37 @@ public void close() {
         }
     }
 
+
+    /**
+     * Creates a Pravega instance.
+     */
+    private static class PravegaRunner {
+        private final int containerCount;
+        private BookKeeperRunner bookKeeperRunner;
+        private SegmentStoreRunner segmentStoreRunner;
+        private ControllerRunner controllerRunner;
+
+        PravegaRunner(int instanceId, int bookieCount, int containerCount, StorageFactory storageFactory) throws Exception {
+            this.containerCount = containerCount;
+            this.bookKeeperRunner = new BookKeeperRunner(instanceId, bookieCount);
+            this.segmentStoreRunner = new SegmentStoreRunner(storageFactory, null, containerCount);
+            this.controllerRunner = new ControllerRunner(this.bookKeeperRunner.bkPort, this.segmentStoreRunner.servicePort,
+                    containerCount);
+        }
+
+        public void restartControllerAndSegmentStore(StorageFactory storageFactory, BookKeeperLogFactory dataLogFactory)
+                throws DurableDataLogException, InterruptedException {
+            this.segmentStoreRunner = new SegmentStoreRunner(storageFactory, dataLogFactory, this.containerCount);
+            this.controllerRunner = new ControllerRunner(this.bookKeeperRunner.bkPort, this.segmentStoreRunner.servicePort, containerCount);
+        }
+
+        public void close() throws Exception {
```

Comment:
You'll need to add an `@Override` tag once you implement `AutoCloseable`

Reply:
Fixed.